components/logins/README.md

```diff
@@ -15,11 +18,70 @@ The relevant directories are as follows:
+
+### Sign-out behavior
+When the user signs out of their Firefox Account, we reset the storage and clear the shared parent.
+
+### Merging records
+When records are added, the logins component performs a three-way merge between the local record, the remote record and the shared parent (last update on the server).  Details on the merging algorithm are contained in the [generic sync rfc](https://github.com/mozilla/application-services/blob/1e2ba102ee1709f51d200a2dd5e96155581a81b2/docs/design/remerge/rfc.md#three-way-merge-algorithm)
+
+### Record de-duplication
+
+De-duplication compares the records for same the username and same url, but with different passwords.  Deduplication logic is based on age, the username and hostname it performs a record update unless it older than the local record. If the change is older than our local records, and you have changed the same field on both, the record is not updated.
+De-duplication compares the records for same the username and same url, but with different passwords.  Deduplication logic is based on age, the username and hostname.
+- If the changes are newer than the local record it performs an update.
+- If the change is older than our local records, and you have changed the same field on both, the record is not updated.
+
+## Getting started
+
+**Prerequisites**: Firefox account authentication is necessary to obtain the keys to decrypt synced login data.  See the [android-components FxA Client readme](https://github.com/mozilla-mobile/android-components/blob/master/components/service/firefox-accounts/README.md) for details on how to implement on Android.  For iOS, Firefox for iOS still implement the legacy oauth.
+
+**Platform-specific details**:
+- Android: add support for the logins component via android-components [Firefox Sync - Logins](https://github.com/mozilla-mobile/android-components/blob/master/components/service/sync-logins/README.md) service.
+- iOS: start with the [guide to consuming rust components on iOS](https://github.com/mozilla/application-services/blob/master/docs/howtos/consuming-rust-components-on-ios.md)
+
+## API Documentation
+- TODO [Expand and update API docs](https://github.com/mozilla/application-services/issues/1747)
+
+## Testing
+
+![status-img](https://img.shields.io/static/v1?label=test%20status&message=acceptable&color=darkgreen)
+
+Our goal is to seek an _acceptable_ level of test coverage, which basically means we feel confident that if we are making changes to the component the testing will alert us to related regressions.  When making changes in an area, make an effort to improve (or minimally not reduce) coverage. Test coverage assessment includes:
+* [rust tests](https://github.com/mozilla/application-services/blob/master/testing/sync-test/src/logins.rs)
+* TODO add link [android tests]()
+* TODO add link [ios tests]()
+* TODO add link [ffi tests]
```

Comment:
No specific FFI tests exist (they'd be much too difficult and error-prone to write -- I don't think we'd want these even if we have infinite time to author tests), but the FFI is well-covered by the android tests.